### ISSUE TITLE
TinyMCE crashes Chromium, Firefox on &lt; character

### ISSUE THREAD

**Issue by CatalinFrancu** [tinymce/tinymce#3047]
Sorry to file the umpteenth crash report. I have read several others and I believe this one is different.

The following (valid) HTML file will crash the tab in Chromium and Firefox. Note that the "less than" sign inside the textarea is properly escaped as `&lt;`. I can make it work by (1) inserting a space immediately after the `&lt;` or (2) escaping the ampersand itself, like `&amp;lt;`

```
<!DOCTYPE html>
<html>
  <head>
    <meta charset="utf-8"/>
    <script src="https://code.jquery.com/jquery-2.2.4.min.js"></script>
    <script src="http://cdn.tinymce.com/4/tinymce.min.js"></script>
    <script>
      $(function() {
        tinymce.init({
          selector: 'textarea',
        });
      });
    </script>
    <title>title</title>
  </head>

  <body>
    <form>
      <textarea>
        test &lt;test test test test test test test test test test test test test 
      </textarea>
    </form>
  </body>
</html>
```

Thanks for TinyMCE!

**Comment by miglesiasEB** [comment]
It happened to me too on the HTML editor.

Pasting this:
```
<span class="zw-portion" style="box-sizing: border-box; font-family: arial, helvetica, sans-serif; font-size: 12pt;"
```
crashes the browsers, pasting this:
```
< span class="zw-portion" style="box-sizing: border-box; font-family: arial, helvetica, sans-serif; font-size: 12pt;"
```

doesn't.

**Comment by jayarjo** [comment]
Related: https://github.com/tinymce/tinymce/issues/3300

**Comment by justinh-eb** [comment]
It seems that the issue is the html parser's regular expression ends up backtracking catastrophically if it believes what it is matching on is an incomplete tag. That's why the space after the `<` keeps the browser from crashing. In the case that @miglesiasEB demonstrates, if you remove the contents of the style tag 

Here is a minimal example to demonstrate the issue in the regex itself pulled from TinyMCE v3.5.11:

```
l = new RegExp("<(?:(?:!--([\\w\\W]*?)-->)|(?:!\\[CDATA\\[([\\w\\W]*?)\\]\\]>)|(?:!DOCTYPE([\\w\\W]*?)>)|(?:\\?([^\\s\\/<>]+) ?([\\w\\W]*?)[?/]>)|(?:\\/([^>]+)>)|(?:([A-Za-z0-9\\-\\:\\.]+)((?:\\s+[^\"'>]+(?:(?:\"[^\"]*\")|(?:'[^']*')|[^>]*))*|\\/|\\s+)>))","g");

E = '<span class="zw-portion" style="box-sizing: border-box; font-family: arial, helvetica, sans-serif;" data-format="{"fv":"normal","cs":"normal","fw":"normal","fgc":"rgb(0, 0, 0)","size":12,"type":"text","fs&qu"'

l.exec(E)
```

**Comment by lnewson** [comment]
I was able to reproduce this in TinyMCE 4.5.11, however I'm unable to reproduce it in the latest version of TinyMCE. Here's a fiddle to show it working: http://fiddle.tinymce.com/Oghaab

It's likely this has been fixed somewhere between 4.5.11 and 4.9.10, so I'm going to close this issue. If you're still able to reproduce this in the latest version however, then let us know and we'll reopen this issue.

Thanks!
